streams/src/main/java/org/apache/kafka/streams/kstream/ValueTransformer.java

```diff
@@ -62,6 +62,8 @@
      * {@code ValueTransformer} and will result in an {@link StreamsException exception}.
      *
      * @param context the context
+     * @throws IllegalStateException If store get's registered after initialization is already finished
```

Comment:
Is this really some error that users can (even intentionally) make? I felt if it ever happens, it should be a bug in our own code as users cannot really achieve this scenario from the public APIs right?

Reply:
One possibility that I can think of, is that users call `stream.start()`, and then call `register` in some context. That sounds too unlikely to me, but if you feel strongly about this, feel free to keep it as is.

Ditto for all other places.

Reply:
I agree, that this should indicate an internal bug (not even sure, if the second scenario you describe could happen). I only document what get's thrown here to make exception tracking simpler. We should revisit this, but not as part of this PR. WDYT about this proposal?